src/routes/Diagram.ts

```diff
@@ -1,47 +1,36 @@
-import { Enumeration, OpenAPIRoute, OpenAPISchema, Query, Str } from '@cloudflare/itty-router-openapi'
+import { Enumeration, OpenAPIRoute, Query, Str, OpenAPISchema } from '@cloudflare/itty-router-openapi'
 
 import { saveShortLink } from './Shorten'
 
 import { createTrackerForRequest, sendMixpanelEvent } from '../mixpanel'
 import { Env } from '..';
 import { diagramDetails } from "./diagrams";
-import { DiagramLanguage, diagramLanguages } from "./diagrams/utils";
-
-// TODO: Add graphvis editor https://www.devtoolsdaily.com/graphviz/?#%7B%22dot%22%3A%22digraph%20MessageArchitecture%20%7B%5Cn%20%20messageClient%5Cn%20%20messageQueue%5Bshape%3Drarrow%5D%5Cn%7D%22%7D
+import { DiagramLanguage, diagramLanguages, DiagramType, diagramTypes } from "./diagrams/utils";
 
 export class MermaidRoute extends OpenAPIRoute {
   /// 2. Creates /openapi.json route under the hood. Injects this into gpt prompt to teach about how to use the plugin.
   static schema = {
     tags: ["Diagram"],
-    summary: `Taking in a mermaid graph diagram, renders it and returns a link to the rendered image.`,
+    summary: `Taking a diagram, renders it and returns a link to the rendered image.`,
     parameters: {
       // TODO: Pass manifest version as a parameter so its easier to debug old / new clients
       // not sure if its even possible.
 
       // It still prefers the old name even if new manifest is fetched
-      mermaid: Query(
-        new Str({
-          description: "Mermaid to render (legacy parameter name, use diagram instead)",
-          // Not providing an example because it's a duplicate of the one below
-        }),
-        {
-          required: false,
-        }
-      ),
       diagram: Query(
         new Str({
           description: "Diagram to render",
           example: `graph TB\\n  U[\\"User\\"] -- \\"File Operations\\" --> FO[\\"File Operations\\"]\\n  U -- \\"Code Editor\\" --> CE[\\"Code Editor\\"]\\n  FO -- \\"Manipulation of Files\\" --> FS[\\"FileSystem\\"]\\n  FS -- \\"Write/Read\\" --> D[\\"Disk\\"]\\n  FS -- \\"Compress/Decompress\\" --> ZL[\\"ZipLib\\"]\\n  FS -- \\"Read\\" --> IP[\\"INIParser\\"]\\n  CE -- \\"Create/Display/Edit\\" --> WV[\\"Webview\\"]\\n  CE -- \\"Language/Code Analysis\\" --> VCA[\\"VSCodeAPI\\"]\\n  VCA -- \\"Talks to\\" --> VE[\\"ValidationEngine\\"]\\n  WV -- \\"Render UI\\" --> HC[\\"HTMLCSS\\"]\\n  VE -- \\"Decorate Errors\\" --> ED[\\"ErrorDecoration\\"]\\n  VE -- \\"Analyze Document\\" --> TD[\\"TextDocument\\"]\\n`,
         }),
         {
-          required: false,
+          required: true,
```

Comment:
We should not require any of the parameters. These should be backwards compatible. Thats the reason I made them not required at the moment.

## Consider this scenario:
User A installs Show Me - they get the openapi definition (will send mermaid=...)
We deploy this newer version to the server (makes diagram required)
User A send the request to our server (mermaid=...)
Server sends an error back because OpenAPI helper class performs validation against its schema - and it didn't find the required diagram parameter along with others.

## Backwards compatible change:
We simply add new parameters keeping the optional
We also continue to handle requests from old clients

Reply:
You can reproduce by checking out main, installing the plugin in dev mode, then switching to this branch and rebuilding the server. Then run the plugin: 
<img width="313" alt="image" src="https://github.com/bra1nDump/show-me-chatgpt-plugin/assets/12608159/c169485a-b7ad-4f06-b3e2-7c1cb17f80d3">

Reply:
Got it! I will make it not required